src/cpp/src/perf_metrics.cpp

```diff
@@ -95,14 +95,16 @@ void PerfMetrics::evaluate_statistics(std::optional<TimePoint> start_time) {
     // If start_item is specified then recalculate durations according to start times and calculate statistics only after that.
     if (start_time.has_value() && raw_metrics.m_new_token_times.size() > 0 && raw_metrics.m_batch_sizes.size() > 0) {
         auto start_time_val = *start_time;
-        auto& tok_times = raw_metrics.m_new_token_times;
-        auto& batch_sizes = raw_metrics.m_batch_sizes;
+        const auto& tok_times = raw_metrics.m_new_token_times;
+        const auto& batch_sizes = raw_metrics.m_batch_sizes;
+
+        raw_metrics.m_durations.clear();
         raw_metrics.m_durations.reserve(tok_times.size());
 
         auto ttft = tok_times[0] - start_time_val;
-        raw_metrics.m_times_to_first_token = std::vector<MicroSeconds>();
+        raw_metrics.m_times_to_first_token.clear();
         raw_metrics.m_times_to_first_token.emplace_back(ttft);
-        num_generated_tokens = batch_sizes[0];
+        num_generated_tokens = 0;
```

Comment:
Is it ok to skip tokens generated in first iteration?
Below we exclude `prefil` stage, so `batch[0]` seems to be not summed up
`for (size_t i = 1; i < tok_times.size(); ++i) {`

Reply:
reverted and fixed CB's filling of `m_batch_sizes` value